otx/algorithms/segmentation/adapters/mmseg/task.py

```diff
@@ -0,0 +1,607 @@
+    FeatureVectorHook,
+)
+from otx.algorithms.common.adapters.mmcv.utils import (
+    build_data_parallel,
+    get_configs_by_pairs,
+    patch_data_pipeline,
+    patch_from_hyperparams,
+)
+from otx.algorithms.common.adapters.mmcv.utils.config_utils import (
+    MPAConfig,
+    update_or_add_custom_hook,
+)
+from otx.algorithms.common.configs.training_base import TrainType
+from otx.algorithms.common.utils import set_random_seed
+from otx.algorithms.common.utils.data import get_dataset
+from otx.algorithms.common.utils.logger import get_logger
+from otx.algorithms.segmentation.adapters.mmseg.configurer import (
+    IncrSegmentationConfigurer,
+    SegmentationConfigurer,
+    SemiSLSegmentationConfigurer,
+)
+from otx.algorithms.segmentation.adapters.mmseg.utils.builder import build_segmentor
+from otx.algorithms.segmentation.adapters.mmseg.utils.exporter import SegExporter
+from otx.algorithms.segmentation.task import OTXSegmentationTask
+
+# from otx.algorithms.segmentation.utils import get_det_model_api_configuration
+from otx.api.configuration import cfg_helper
+from otx.api.configuration.helper.utils import ids_to_strings
+from otx.api.entities.datasets import DatasetEntity
+from otx.api.entities.inference_parameters import InferenceParameters
+from otx.api.entities.model import (
+    ModelEntity,
+    ModelFormat,
+    ModelOptimizationType,
+    ModelPrecision,
+)
+from otx.api.entities.subset import Subset
+from otx.api.entities.task_environment import TaskEnvironment
+from otx.api.serialization.label_mapper import label_schema_to_bytes
+from otx.api.usecases.tasks.interfaces.export_interface import ExportType
+from otx.core.data import caching
+
+logger = get_logger()
+
+RECIPE_TRAIN_TYPE = {
+    TrainType.Semisupervised: "semisl.py",
+    TrainType.Incremental: "incremental.py",
+    TrainType.Selfsupervised: "selfsl.py",
+}
+
+# TODO Remove unnecessary pylint disable
+# pylint: disable=too-many-lines
+
+
+class MMSegmentationTask(OTXSegmentationTask):
+    """Task class for OTX segmentation using mmsegmentation training backend."""
+
+    # pylint: disable=too-many-instance-attributes
+    def __init__(self, task_environment: TaskEnvironment, output_path: Optional[str] = None):
+        super().__init__(task_environment, output_path)
+        self._data_cfg: Optional[Config] = None
+        self._recipe_cfg: Optional[Config] = None
+
+    # pylint: disable=too-many-locals, too-many-branches, too-many-statements
+    def _init_task(self, export: bool = False):  # noqa
+        """Initialize task."""
+
+        if (
+            self._train_type in RECIPE_TRAIN_TYPE
+            and self._train_type == TrainType.Incremental
+            and self._hyperparams.learning_parameters.enable_supcon
+            and not self._model_dir.endswith("supcon")
+        ):
+            self._model_dir: str = os.path.join(self._model_dir, "supcon")
+            self.data_pipeline_path: str = os.path.join(self._model_dir, "data_pipeline.py")
```

Comment:
Shouldn't this be in `_init_`?

Reply:
`__init__` already defines `self._model_dir` and `self._data_pipeline_path`, but supcon requires redefinition of both two attributes to find supcon recipe.

Reply:
Fixed, thank you for sharing good idea!